tailnet/configmaps.go

```diff
@@ -542,42 +610,81 @@ func (c *configMaps) fillPeerDiagnostics(d *PeerDiagnostics, peerID uuid.UUID) {
 	if !ok {
 		return
 	}
+
 	d.ReceivedNode = lc.node
-	ps, ok := status.Peer[lc.node.Key]
+	if lc.node != nil {
+		ps, ok := status.Peer[lc.node.Key]
+		if !ok {
+			return
+		}
+		d.LastWireguardHandshake = ps.LastHandshake
+	}
+	return
+}
+
+func (c *configMaps) peerReadyForHandshakeTimeout(peerID uuid.UUID) {
+	c.L.Lock()
+	defer c.L.Unlock()
+	lc, ok := c.peers[peerID]
 	if !ok {
 		return
 	}
-	d.LastWireguardHandshake = ps.LastHandshake
+	if lc.readyForHandshakeTimer != nil {
```

Comment:
I don't think you need to check for the timer being non-nil, and I don't think you need to clear it.